File: sort/quicksort.py
def quicksort(list,low,high):
    if low < high:
        s = partition(list,low,high)
        quicksort(list,low,s-1)
        quicksort(list,s+1,high)

def partition(list,low,high):
    i = low
    j = high
    while i<j:
        while i<high and list[low]>=list[i]:
            i = i+1
        while low<=j and list[low]<list[j]:
            j = j-1
        temp = list[i]
        list[i] = list[j]
        list[j] = temp
    temp = list[i]
    list[i] = list[j]
    list[j] = temp
    temp = list[j]
    list[j] = list[low]
    list[low] = temp
    return j

#a = [3,4,2,0,8,9,1,7,6,5]
#b = [5,2,4,6,5,7,3,1,2,0]
#quicksort(a,0,len(a)-1)
#print (a)
#quicksort(b,0,len(b)-1)
#print (b)

File: JAVA/basicAlgrithms/QuickSort.java
package algorithm;

public class QuickSort {
	public static int partition(int[] array, int low, int high) {
		int i = low;
		int j = high;
		while (i < j) {
			while (i < high && array[low] >= array[i]) {
				i++;
				// System.out.print(i + " ");
			}
			while (j >= low && array[low] < array[j]) {
				j--;
				// System.out.println(j + " ");
			}
			swap(array, i, j);
		}

		swap(array, i, j);
		swap(array, low, j);
		return j;
	}

	public static void swap(int[] array, int a, int b) {
		int temp = array[a];
		array[a] = array[b];
		array[b] = temp;
	}

	public static void quickSort(int[] array, int low, int high) {
		if (low < high) {
			int pivit = partition(array, low, high);
			quickSort(array, low, pivit - 1);
			quickSort(array, pivit + 1, high);

		}

	}

	public static void main(String[] args) {
		int[] array = new int[] { 3, 59, 7, 2, 4, 9, 0, 5, 4, 4, 2, 56, 33, 21 };
		quickSort(array, 0, array.length - 1);
		for (int i : array) {
			System.out.print(i + " ");
		}
	}
}
